### PR TITLE
Update to Geth v1.13.1 (Zakros) - [depends on #136 v1.12.2 update]

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

Update to include v1.13.1 changes of upstream geth.

Notable changes:
- Block-building params are finally in a struct, making it more easy to extend
- Engine API payload attributes now have optional beacon block root
- Moved engine API heartbeat to early-exit out of it, rather than not starting the go-routine, since the usage of the routine moved upstream.
- No more txpool tx wrapper (back to sidecar pattern for blob tx)
- The TDD-Merged checks were removed, assumed post-merge readiness now that PoW is gone.
- `rlpLog` functionality is now just part of `Log`
- Block builder now supports EIP-4844 (blob txs) and EIP-4788 (beacon block root).

This depends on #136 to be merged first, to land the v1.12.2 changes.

### CODE REVIEW

**Comment by protolambda** [comment]
v1.12.2 changes landed in `optimism` branch now, updated the branch to handle the fork-override and v1.12.2 review changes, so v1.13.1 is ready for review now.

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -1697,16 +1726,11 @@ func CheckExclusive(ctx *cli.Context, args ...interface{}) {
 // SetEthConfig applies eth-related command line flags to the config.
 func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 	// Avoid conflicting network flags
-	CheckExclusive(ctx, MainnetFlag, DeveloperFlag, GoerliFlag, SepoliaFlag, BetaOPNetworkFlag)
+	CheckExclusive(ctx, MainnetFlag, DeveloperFlag, GoerliFlag, SepoliaFlag, HoleskyFlag, BetaOPNetworkFlag)
```
Comment (tynes) [inline cmd/utils/flags.go:1729]: Will want to follow up and remove the beta prefix?

**Review by tynes — APPROVED** [review]
This is quite large but i looked through a bunch of it, assuming that its good if tests are passing

**Review by Inphi — APPROVED** [review]
LGTM!
To note for the op-node changes; withdrawals is now required a required attribute, even if empty, for the engine API.

File: cmd/utils/flags.go (hunk 1 above)
Comment (protolambda) [reply cmd/utils/flags.go:1729]: We have to keep the beta prefix on the network flag until the status of the network options clears up I believe

**Pull request merged**